pkg/virt-launcher/virtwrap/api/converter.go

```diff
@@ -992,6 +990,9 @@ func calculateRequestedVCPUs(vmi *v1.VirtualMachineInstance) uint32 {
 	if vmi.Spec.Domain.CPU != nil && vmi.Spec.Domain.CPU.Cores != 0 {
 		return vmi.Spec.Domain.CPU.Cores
 	}
+	if !vmi.IsCPUDedicated() {
```

Comment:
Hm I believe we must set defaults under mutating webhook, so we should not get the situation when `vmi.Spec.Domain.CPU == nil || vmi.Spec.Domain.CPU.Cores == 0`